### PR TITLE
Show delaying and tracing in Emacs mode line

### PR DESCRIPTION
**Pull request by brandm**
This is convenient to see what happens during edit and update when using or developing for the Emacs plugin. Test with something slow like `time.sleep(1.4)`.

### CODE REVIEW

**Comment by donkirkby** [comment]
Thanks for the pull request, it looks like you've done some good clean up of the elisp code.
I can see how the minor mode suffix of "-D", "-T", or "-t" would be helpful when working on the Emacs mode, but I think it's just clutter for regular users. Maybe there can be a switch variable to turn it on during development. Also, I'm not sure that it's actually updating the display when you change the lighter variable. I sometimes saw a long delay before the "-t" suffix appeared.
Finally, the delay helps avoid locking the editor, but it still locks when the user stops typing. I wonder if we could use a [filter function] to receive the results from the tracing process, so it wouldn't have to block while waiting for the results.

In summary, I'll happily merge this if you remove or disable the suffixes. If you want to try a filter function, that would be awesome.

[filter function]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Filter-Functions.html

**Comment by brandm** [comment]
Thanks for mentioning the filter function, I will have a look later. Less locking would be very helpful.

I thought about a configurable lighter before so that the different strings for the different states are configurable with "live-D", "live-T", "live-t" and "live-FAIL" as one of the possibilities. Now I will do that.

Can you try if adding a `force-mode-line-update` after setting the lighter to "live-t" fixes the lagging update you mentioned?

**Comment by brandm** [comment]
> I thought about a configurable lighter before so that the different strings for the different states are configurable with "live-D", "live-T", "live-t" and "live-FAIL" as one of the possibilities. Now I will do that.

Assuming that the Travis CI build failure was not caused by me this updated branch is ready for review and merge.

**Comment by donkirkby** [comment]
The failure looks like it was caused by a service outage. It worked fine after a retry.

Thanks for all the improvements.
